yt_dlp/extractor/mediastream.py

```diff
@@ -2,16 +2,43 @@
 
 from .common import InfoExtractor
 from ..utils import (
+    clean_html,
     remove_end,
     str_or_none,
-    strip_or_none,
     traverse_obj,
     urljoin,
 )
 
 
-class MediaStreamIE(InfoExtractor):
-    _VALID_URL = r'https?://mdstrm.com/(?:embed|live-stream)/(?P<id>\w+)'
+class MediaStreamBaseIE(InfoExtractor):
+    _EMBED_BASE_URL = 'https://mdstrm.com/embed'
+    _BASE_URL_RE = r'https?://mdstrm\.com/(?:embed|live-stream)'
+
+    def _extract_mediastream_urls(self, webpage):
```

Comment:
If you can make this a `classmethod`, then it can be defined as `MediaStreamIE_extract_embed_urls`. Base-class and _extract_from_webpage aren't needed then.

Reply:
I don't know if I can. It calls `self._yield_json_ld`, which is why I went with the base class/_extract_from_webpage idea.

If we could generally solve the JSON LD `embedUrl` problem, I think this would be moot. But these mediastream sites don't make it easy, since on some of these pages `contentUrl` is a direct link to a low-quality mp4, and `embedUrl` is the link we need for HLS+DASH+metadata. And then other times, `embedUrl` is `"0"` and `contentUrl` is the one we need. (and some pages have no JSON LD URLs at all)